lib/bundlex.ex

```diff
@@ -6,39 +6,73 @@ defmodule Bundlex do
   alias Bundlex.Helper.MixHelper
   alias Bundlex.Platform
 
-  @type platform_t :: :linux | :macosx | :windows32 | :windows64 | :nerves
+  @type platform_t :: :linux | :macosx | :windows32 | :windows64 | :nerves | :custom
 
   @typedoc """
-  A map containing four fields that describe the platform.
+  A map containing four fields that describe the target platform.
 
   It consists of:
   * architecture - e.g. `x86_64` or `arm64`
   * vendor - e.g. `pc`
   * os - operating system, e.g. `linux` or `darwin20.6.0`
-  * abi - application binary interface, e.g. `musl` or `gnu` (nil if unknown / non-existent)
+  * abi - application binary interface, e.g. `musl` or `gnu`
   """
   @type target ::
-          %{architecture: String.t(), vendor: String.t(), os: String.t(), abi: String.t() | nil}
+          %{
+            architecture: String.t() | :unknown,
+            vendor: String.t() | :unknown,
+            os: String.t() | :unknown,
+            abi: String.t() | :unknown
+          }
 
   @doc """
-  A function returning a target triplet for the environment on which it is run.
+  A function returning information about the target platform. In case of cross-compilation the
+  information can be provided by setting appropriate environment variables.
   """
   @spec get_target() :: target()
-  def get_target() do
-    [architecture, vendor, os | maybe_abi] =
-      :erlang.system_info(:system_architecture) |> List.to_string() |> String.split("-")
-
-    %{
-      architecture: architecture,
-      vendor: vendor,
-      os: os,
-      abi: List.first(maybe_abi)
-    }
+  case System.fetch_env("CROSSCOMPILE") do
```

Comment:
This will be only executed when compiling bundlex

Reply:
That was my goal, I believe it worked the same way with mix target

Reply:
Yeah, the problem is that we were sure that bundlex is recompiled when the target changes